containers/src/Kokkos_UnorderedMap.hpp

```diff
@@ -818,7 +818,7 @@ class UnorderedMap {
   bool m_bounded_insert;
   hasher_type m_hasher;
   equal_to_type m_equal_to;
-  mutable size_type m_size;
+  mutable Kokkos::View<size_type, Kokkos::HostSpace> m_size;
```

Comment:
Is `mutable` still necessary?

Reply:
I would lean toward removal.

Since this is strictly host-space, the meaning could be clarified by making it `shared_ptr<size_type>` instead of using `View` as such.

Reply:
Yeah, let's try that.